packages/grid/_modules_/grid/hooks/features/pagination/gridPaginationReducer.ts

```diff
@@ -88,7 +88,7 @@ export const setGridRowCountStateUpdate = (state, payload): GridPaginationState
       ...state,
       pageCount: newPageCount,
       rowCount: totalRowCount,
-      page: state.page > newPageCount ? newPageCount : state.page,
+      page: state.page > newPageCount ? newPageCount - 1 : state.page,
```

Comment:
This is not correct. See https://codesandbox.io/s/material-demo-forked-or2s9?file=/demo.js, it doesn't work. I'm sending another PR